src/OrbitBase/CMakeLists.txt

```diff
@@ -71,6 +71,7 @@ target_sources(OrbitBase PRIVATE
 endif()
 
 target_link_libraries(OrbitBase PUBLIC
+        ApiInterface
```

Comment:
Such a dependency for `OrbitBase` is a bit worrisome. What's the reason for it? I assume because of `Tracing.cpp` (i.e., introspection). But in that case I think that `Tracing.cpp` should be moved to its own module, that modules that use introspection can explicitly depend on. Or introspection and manual instrumentation should just be independent.

Reply:
This is an improvement over what we had before (including the header user "../../src/Orbit.h" but I agree that introspection should probably be in its own module. Added b/182473973

Reply:
It's an improvement in the sense that we don't just include a header bypassing CMake, but it's also quite worse because now we explicitly have `OrbitBase` depending on one of our modules. We should not introduce exceptions to the "rule" that `OrbitBase` shouldn't depend on other modules. In my opinion this should be addressed, creating a bug doesn't really do anything.

Reply:
The dependency was already there, this is making it explicit. OrbitBase itself uses introspection btw, which make that "rule" difficult to enforce. Note that introspection is not only using Orbit.h, but also EncodedEvent.h. I made a bug because I think it's a significant change that goes out of the scope of this PR. It also ties in with the deprecation of the old API that will allow us to remove EncodedEvent.h altogether. I think it will be easier to tackle this then.

Reply:
> OrbitBase itself uses introspection btw, which make that "rule" difficult to enforce.

If something is already using introspection in OrbitBase then ok, this goes beyond this change. I would at least add a todo referring to the bug though. But in my opinion OrbitBase should not use introspection. That goes completely against OrbitBase being a "base" component, i.e., independent of Orbit's technicalities, and it introduces unwanted dependencies.